### PR TITLE
Added a page in installer for express/custom installation

### PR DESCRIPTION
**Pull request by utkarshgautamnyu**
Added a page in the installer to choose express/custom installation type 
Express install -> No Post Install Options provided to user. Software installed with default options. 
Custom Install -> Post Install Options provided to user

Testing Notes:
1) Build the PR
2) Generate the installer following these documents: https://github.com/highfidelity/hifi/blob/master/INSTALL.md , #6870
3) Run the installer
4) Observe a new page - "Choose Installation Type"
   a) "Express Install" is selected by default
   b) Lower button panel has labels: Back, Install, Cancel
   c) Click on "Custom Install"
   d) Button panel changes to Back, Next, Cancel
5) Install with Express install selected
   a) No Post Install Options page
   b) Software installs with following defaults: 
       Create a desktop shortcut for Interface
       Launch Sandbox after install
       Launch Interface after install
       Launch Sandbox on startup
6) Install with Custom Install selected
   a) Post Install options page appears
   b) If it is a PR build, Observe that checkbox for copying settings from Production is unchecked by default   
   c) Change any of the default settings (For example uncheck Desktop shortcut creation)
   d) Observe the settings are reflected (For example Desktop shortcut is not created)
7) Come back and Install again with "Express Install" selected
   a) Observe changes made in custom install (For example uncheck Desktop shortcut creation), are over ridden by defaults, i.e Desktop shortcut IS created
8) PR build tests:
    a) In custom install, select the "copy from production" checkbox and install
    b) Come back and install with Express Install, it **Does Not** copy settings/content from production

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1791/HighFidelity-Beta-PR11004-0a64a1e3b0dff67cf5c1452cbe659f25bc399145.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1791/HighFidelity-Beta-PR11004-0a64a1e3b0dff67cf5c1452cbe659f25bc399145.exe

**Comment by YuppyPlays** [comment]
Testing with Gustavo build (if this is okay).

**Comment by utkarshgautamnyu** [comment]
I've made some minor changes, can you wait for them to pass before testing. Also you'd have to generate the installer as mentioned in the docs

**Comment by YuppyPlays** [comment]
@utkarshgautamnyu   Ah I see, I may leave this for another tester as I'm not experienced in building installer from scratch. However, Gustavo is showing the different installer options as described thus far.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1808/HighFidelity-Beta-PR11004-0277f6a7e6e6b4a7979f5cc13cf73874738ef60a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1809/HighFidelity-Beta-PR11004-de5984d121d3cd05bd39c84e1d78f0aa164f6dbb.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1810/HighFidelity-Beta-PR11004-2310ecab8d88680056af867364493d33a1a105d7.dmg

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1812/HighFidelity-Beta-PR11004-2310ecab8d88680056af867364493d33a1a105d7.exe

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
Testing complete:

Build the PR **Used Gustavo, dev confirmed this is okay**
//Generate the installer following these documents: //https://github.com/highfidelity/hifi/blob/master/INSTALL.md , #6870
Run the installer **Check**
Observe a new page - "Choose Installation Type" **Check**
a) "Express Install" is selected by default **Check**
b) Lower button panel has labels: Back, Install, Cancel **Check**
c) Click on "Custom Install" **Check**
d) Button panel changes to Back, Next, Cancel **Check**
Install with Express install selected **Check**
a) No Post Install Options page **Check**
b) Software installs with following defaults: **Check**
Create a desktop shortcut for Interface **Check**
Launch Sandbox after install **Check**
Launch Interface after install **Check**
Launch Sandbox on startup **Check**
If it is a PR build, copy settings/content from production **Check, retained settings correctly**
Install with Custom Install selected **Check**
a) Post Install options page appears **Check, previous had no extra options, this one shows the ones expected**
b) Change any of the default settings (For example uncheck Desktop shortcut creation) **Check**
c) Observe the settings are reflected (For example Desktop shortcut is not created) **Check**
Come back and Install again with "Express Install" selected **Check**
a) Observe changes made in custom install (For example uncheck Desktop shortcut creation), are over ridden by defaults, i.e Desktop shortcut IS created **Check**

This is a full **Pass** :)

**Comment by birarda** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1821/HighFidelity-Beta-PR11004-2310ecab8d88680056af867364493d33a1a105d7.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1821/HighFidelity-Beta-PR11004-2310ecab8d88680056af867364493d33a1a105d7.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1824/HighFidelity-Beta-PR11004-0ae8f4526e8074d6d952e20b7a830472e60e9482.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1824/HighFidelity-Beta-PR11004-0ae8f4526e8074d6d952e20b7a830472e60e9482.exe

**Comment by YuppyPlays** [comment]
Testing [as per dev request]

**Comment by YuppyPlays** [comment]
Testing complete:

[Using new provided steps]

Build the PR **Using Gustavo build as dev has confirmed this is okay**
//Generate the installer following these documents: //https://github.com/highfidelity/hifi/blob/master/INSTALL.md , #6870
Run the installer **Check**
Observe a new page - "Choose Installation Type" **Check**
a) "Express Install" is selected by default **Check**
b) Lower button panel has labels: Back, Install, Cancel **Check**
c) Click on "Custom Install" **Check**
d) Button panel changes to Back, Next, Cancel **Check**
Install with Express install selected **Check**
a) No Post Install Options page **Check**
b) Software installs with following defaults: **Check**
Create a desktop shortcut for Interface **Check**
Launch Sandbox after install **Check**
Launch Interface after install **Check**
Launch Sandbox on startup **Check**
Install with Custom Install selected **Check**
a) Post Install options page appears **Check**
b) If it is a PR build, Observe that checkbox for copying settings from Production is unchecked by default **Check**
c) Change any of the default settings (For example uncheck Desktop shortcut creation) **Check**
d) Observe the settings are reflected (For example Desktop shortcut is not created) **Check**
Come back and Install again with "Express Install" selected **Check**
a) Observe changes made in custom install (For example uncheck Desktop shortcut creation), are over ridden by defaults, i.e Desktop shortcut IS created **Check**
PR build tests:
a) In custom install, select the "copy from production" checkbox and install **Check**
b) Come back and install with Express Install, it Does Not copy settings/content from production **Check**

**Pass**

Notes: Last time tested, using express still brought up a blank screen with "Install" at the bottom; as part of overriding the options, though the page was redundant and dev has removed and fixed this, streamlined better and will help save a bit of confusion for new users :)

File: cmake/templates/NSIS.template.in [hunk 1]
```diff
@@ -567,6 +624,16 @@ Var LaunchServerNowState
...
 Var CopyFromProductionState
 Var CleanInstallState
+Var ExpressInstallState 
+Var CustomInstallState
+
+Function ReadInstallTypes
+${If} ${SectionIsSelected} ${@CLIENT_COMPONENT_NAME@}
+    ; check if the user asked for express/custom install
+    ${NSD_GetState} $ExpressInstallRadioButton $ExpressInstallState
+    ${NSD_GetState} $CustomInstallRadioButton $CustomInstallState
```
Comment (birarda) [inline cmake/templates/NSIS.template.in]: The rest of the file is 2 spaces right? Can you fix the indentation here?

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1968/HighFidelity-Beta-PR11004-e5ddded464ed54a4f942807e090fa5408eaf6129.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1968/HighFidelity-Beta-PR11004-e5ddded464ed54a4f942807e090fa5408eaf6129.dmg

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
Testing complete:

As discussed with dev, tested this against the previous test plan alongside this flow chart with the exception of the 3rd from left screen in-image is actually in the bottom right area of the flowchart now.
installationuserflowv2


**Pass, All functions all work as the flowchart shows, component changes also work for example choosing only selecting Interface to install will only bring up interface related tickboxes after. Also noticed the clean install option which has been placed and confirmed working. All looks good here!**

**Pull request merged**